### PR TITLE
Describe join pushdowns prioritizing correctness over performance

### PR DESCRIPTION
**Pull request by jhlodin**
## Description

Provide more detail around pushdown decisions.

## Additional context and related issues

In order to better set expectations around when pushdown can occur (or not occur) but without going unnecessarily in depth around every pushdown decision, it may be beneficial to point out that connectors prioritize correctness over performance.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by wendigo** [comment]
@jhlodin this actuslly applies to all pushdowns types supported (predicate, aggregate, join etc)

**Comment by mosabua** [comment]
> @jhlodin this actually applies to all pushdowns types supported (predicate, aggregate, join etc)

Then we should lift it up into the pushdown page .. might hav to do some more updates there @wendigo and team can help with that

**Comment by jhlodin** [comment]
@mosabua @wendigo I can lift it up, what's missing is some generic content up there that talks about how pushdown can improve performance. Otherwise this note makes no sense in context, since right now that content only lives in the cost-based join pushdown section.

**Comment by mosabua** [comment]
We should add the relevant higher level info in the pushdown page 

One main differentiating factor for push down is if the underlying data source has any query processing .. object storage connectors can not push down things to process to a system when that system is just files..

File: docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment [hunk 1]
```diff
@@ -10,6 +10,13 @@ improves performance. Note that if no table statistics are available, join
 operation pushdown does not occur to avoid a potential decrease in query
 performance.
 
+.. note::
+
+  The connector performs pushdown where performance may be improved, but in
+  order to preserve correctness an operation may not be pushed down. When
+  pushdown of an operation may result in better performance but risk
+  correctness, the connector prioritizes correctness.
```
Comment (hashhar) [inline docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment:18]: not limited to join, applies to every kind of pushdown - topN, predicate etc.

**Comment by hashhar** [comment]
> object storage connectors can not push down things to process to a system when that system is just files..

Not really true. e.g. in case of Parquet files you can ask the "file" to filter some values out. Even without that in Hive, Iceberg and Delta predicates on partition columns get "pushed down".

File: docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment (hunk 1 above)
Comment (hashhar) [inline docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment:18]:
```suggestion
  pushdown of an operation may result in better performance but risks
  correctness, the connector prioritizes correctness.
```

?

Comment (hashhar) [reply docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment:18]: Can we move this at the beginning of the pushdown section itself? Or a more general page that talks about pushdown?

Comment (jhlodin) [reply docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment]: Moved up (in its own fragment) to the parent section. I don't think it makes sense at the _very_ beginning of the section, but certainly after it's been introduced what the connector's pushdown support is. How's that as a compromise?

**Comment by jhlodin** [comment]
Applied feedback and moved this note up to apply to all pushdown for connectors that support it. Only issue is Delta Lake/Iceberg/Hive don't have dedicated sections describing pushdown so there's not a logical place to put this info. Those connector docs need more work on covering pushdown, so that should probably be a separate effort to this change.

File: docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment (hunk 1 above)
Comment (hashhar) [reply docs/src/main/sphinx/connector/join-pushdown-enabled-true.fragment]: Looks nice. Thanks.

**Review by hashhar — APPROVED** [review]

**Pull request merged**